src/core/src/type/f8e5m2.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/type/f8e5m2.hpp"
+
+#include <cmath>
+#include <limits>
+
+#include "openvino/reference/fake_convert.hpp"
+
+namespace ov {
+static_assert(sizeof(f8e5m2) == 1, "class f8e5m2 must be exactly 1 byte");
+static_assert(std::is_trivially_constructible<f8e5m2, f8e5m2>::value == true);
+static_assert(std::is_trivially_constructible<f8e5m2, ov::float16>::value == false);
+static_assert(std::is_trivially_constructible<f8e5m2, float>::value == false);
```

Comment:
This check not give much. As there are  defined custom ctor this want be trivial constructible. If that condition change to true it would be better (simpler object) but we got compilation error. This should be cheeked for true but I think is not possible.

More important checks:
- `std::is_trivially_destructible` -> the call dtor is not required and memory can be re-used e.g when set_shape in tensor for smaller size.
- is_trivially_copyable